kyuubi-rest-client/src/main/java/org/apache/kyuubi/client/KyuubiRestClient.java

```diff
@@ -0,0 +1,195 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.client;
+
+import java.net.URI;
+import java.util.Base64;
+import javax.net.ssl.SSLContext;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.conn.ssl.SSLConnectionSocketFactory;
+import org.apache.http.conn.ssl.TrustStrategy;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.ssl.SSLContexts;
+import org.apache.kyuubi.client.util.AuthUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KyuubiRestClient {
+
+  private static final Logger LOG = LoggerFactory.getLogger(KyuubiRestClient.class);
+
+  private RestClient httpClient;
+
+  /** Specifies the version of the Kyuubi API to communicate with. */
+  public enum ApiVersion {
+    V1;
+
+    public String getApiNamespace() {
+      return ("api/" + name().toLowerCase());
+    }
+  }
+
+  public enum AuthSchema {
+    BASIC,
+    SPNEGO;
+  }
+
+  public KyuubiRestClient(Builder builder) {
+    // Remove the trailing "/" from the hostUrl if present
+    String hostUrl = builder.hostUrl.replaceAll("/$", "");
+    String baseUrl = String.format("%s/%s", hostUrl, builder.version.getApiNamespace());
+
+    CloseableHttpClient httpclient = initHttpClient(builder);
+
+    String header = initAuthHeader(builder);
+
+    this.httpClient = new RestClient(baseUrl, httpclient, header);
+  }
+
+  public RestClient getHttpClient() {
+    return httpClient;
+  }
+
+  private CloseableHttpClient initHttpClient(Builder builder) {
+    RequestConfig requestConfig =
+        RequestConfig.custom()
+            .setSocketTimeout(builder.socketTimeout)
+            .setConnectTimeout(builder.connectTimeout)
+            .build();
+    SSLConnectionSocketFactory sslSocketFactory;
+    try {
+      TrustStrategy acceptingTrustStrategy = (cert, authType) -> true;
+      SSLContext sslContext =
+          SSLContexts.custom().loadTrustMaterial(null, acceptingTrustStrategy).build();
+      sslSocketFactory = new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE);
+    } catch (Exception e) {
+      LOG.error("Error: ", e);
+      throw new RuntimeException(e);
+    }
+
+    CloseableHttpClient httpclient =
+        HttpClientBuilder.create()
+            .setDefaultRequestConfig(requestConfig)
+            .setSSLSocketFactory(sslSocketFactory)
+            .build();
+    return httpclient;
+  }
+
+  private String initAuthHeader(Builder builder) {
+    String header = "";
+    switch (builder.authSchema) {
+      case BASIC:
+        String account = String.format("%s:%s", builder.username, builder.password);
```

Comment:
account => authorization 


and the password might be null if sasl is enabled.


```
scala> val account = String.forformat("%s:%s", "fwang12", null)
                            ^
       error: value forformat is not a member of object String

scala>
```